Formula/curl.rb

```diff
@@ -47,6 +48,7 @@ def install
       --disable-debug
       --disable-dependency-tracking
       --disable-silent-rules
+      --enable-hsts
```

Comment:
Is this recommended by upstream?

Reply:
> Starting in curl 7.74.0, curl has experimental support for HSTS. Experimental means it isn’t enabled by default and we discourage use of it in production. (Scheduled to be released in December 2020.)

From https://daniel.haxx.se/blog/2020/11/03/hsts-your-curl/

Reply:
We probably shouldn't then.

Reply:
Fantastic!

Reply:
Being experimental doesn't mean it's forbidden to be included in a stable build. After all it's part of a stable release. It means that the _use of the `--hsts` option_ isn't recommended in production. Trying/testing/non-production is not prohibited. In fact, if nobody is able to try it and test it, the likelyhood of ever being recommended for production use is greatly reduced.

/cc @bagder

Reply:
I understand your point of view, and it's probably good for it to be enabled in a tap, or experimental build. But if it is not built by default, we typically tend not to add it.

Upstream *could* have decided to build it by default, but make its use condition on using the `--hsts` option (which is exactly what you seem to suggest). They chose not to, and we should respect their choice.

Reply:
Does this means that in Homebrew all formulae are built using default options?

Turns out this isn't actually so. A `grep -- '--enable'` gives 1047 hits. So I still have to wonder what sort of rule is followed here. Or was all of these 1047 cases carefully followed back to the author's comments to see if the options they enabled are actually recommended in various environments?

The only "burden" for Homebrew maintainters in case of this option, is that in case this feature gets removed and/or finalized in the future, it may need to be deleted from the formulae, worst case. More likely, the option would just become redundant and would not cause further harm until deleted.

Meanwhile it doesn't cause any issues for any existing curl users/uses, but gives a chance to use this feature, for those who are interested.

Just can't spot any loss in this and all this looks like a win-win for everyone to me.

Reply:
@vszakats most of the `--enable` options are simply linked to dependencies. We try to make the best choice in each case, we can't really make everyone happy.

The downside here is clear: if the issue causes trouble, we get blamed, because upstream recommends not to build it, and we did build it.

Reply:
Speaking of a feature that needs to be explicitly enabled by the user to have any effect, this downside appears to be so slim that with that reasoning maybe no project should be included in Homebrew at all, as they may contain a bug, and someone may report that bug to Homebrew directly. I mean, I understand where this comes from, but I feel that Homebrew has quite a hardliner view on this particular case. It's likely that nobody will ever bother any Homebrew maintainer because of this one. If merely enabling it at build-time can have any unwanted side-effects, I'm quite sure it will quickly trigger a bugfix release of `curl`, so this also shouldn't be of special concern for Homebrew maintainers.

Having said that, the [official](https://github.com/curl/curl-for-win) Windows release offered by curl has this option enabled. (Disclaimer: I'm the maintainer of said build.)

Reply:
> this downside appears to be so slim

Then curl should build it by default, if there is no downside. This would have the extra advantage that *all* distributions get it. Think how much better your time is spent convincing upstream than convincing us.